runtime/manifest.js

```diff
@@ -770,7 +770,23 @@ ${e.message}
           let providedSlot = slotConn.providedSlots[ps.param];
           if (providedSlot) {
             if (ps.name) {
-              items.byName.set(ps.name, providedSlot);
+              if (items.byName.has(ps.name)) {
+                // The slot was added to the recipe twice - once as part of the
+                // slots in the manifest, then as part of particle spec.
+                // Unifying both slots, updating name and source slot connection.
+                let theSlot = items.byName.get(ps.name);
+                assert(theSlot !== providedSlot);
+                assert(!theSlot.name && providedSlot);
+                assert(!theSlot.sourceConnection && providedSlot.sourceConnection);
+                assert(theSlot.handleConnections.length == 0);
+                theSlot.name = providedSlot.name;
```

Comment:
`theSlot` (horrible variable name) is already part of the recipe that is being processed. it so happened that it's ben created twice - once because the recipe has `slot '...' as slotX, and second time when particle spec is being set, and this slot is being provided.
The change is that we used NOT to serialize slots that are provided by this recipe, but now (in this PR) we are, in order to get persistent slot IDs.

Another way to handle this (maybe more elegant?) is to pass the `items.byName` map to the particle's `set spec` method (but because it is a setter, i cannot pass additional parameters, and changing that would require many changes in the code, so i'd like to discuss and agree on it first...)

Reply:
Is a similar thing happening with handles? If not, why not?

Reply:
Added an item for it in the Big List ... doc.